lib/net/ssh/transport/algorithms.rb

```diff
@@ -213,22 +213,33 @@ def prepare_preferred_algorithms!
         options[:compression] = %w(zlib@openssh.com zlib) if options[:compression] == true
 
         ALGORITHMS.each do |algorithm, list|
-          algorithms[algorithm] = list.dup
-
           # apply the preferred algorithm order, if any
           if options[algorithm]
-            algorithms[algorithm] = Array(options[algorithm]).compact.uniq
+            option = Array(options[algorithm]).compact.uniq
+
+            if option.first.start_with?('+')
```

Comment:
we either need to handle that option with zero element here or make sure it's not possible on all the code paths calling this.